include/git2/sys/refdb_backend.h

```diff
@@ -251,4 +254,55 @@ GIT_EXTERN(int) git_refdb_set_backend(
 
 GIT_END_DECL
 
+typedef int GIT_CALLBACK(git_refdb_backend_del_v1)(git_refdb_backend *backend,
+					  const char *ref_name,
+					  const git_oid *old_id, const char *old_target);
+
+/* Old v1 struct version */
+struct git_refdb_backend_v1 {
```

Comment:
We still haven't decided how we actually want to proceed with such changes. Thing is, by renaming the old struct we're effectively also breaking old programs, as they cannot be compiled anymore (or worse, the compiler may just print an incompatible function pointer warning that goes unnoticed, and we run into strange behavior). This scenario also isn't helped by GIT_REFDB_BACKEND_VERSION, as a recompile would also get the newer version.

So maybe we need to instead introduce a new `git_refdb_backend_v2`?